From a37935f7c831d654ceef82b2321596ab3d397f94 Mon Sep 17 00:00:00 2001
From: MohammadHossein Qanbari <mohammad.qanbari@qt.io>
Date: Thu, 18 Sep 2025 16:53:41 +0200
Subject: [PATCH 194/553] CRA review: src/widgets/kernel/qapplication.*

The qapplication.* files' security scores are tagged as significant
since the command-line arguments and environment variables are
considered trusted input. It does not expose to high-risk
vulnerabilities and the data is available for validation before
reaching QApplication.

QUIP: 23
Fixes: QTBUG-140349
Task-number: QTBUG-135741
Pick-to: 6.8
Change-Id: Iadb7b4b8f516f89b70999b9ae8993d75084a5adb
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit 33855af7f79ae06a4fbded60b2e88758c7bb0c4a)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/widgets/kernel/qapplication.cpp | 1 +
 src/widgets/kernel/qapplication.h   | 1 +
 src/widgets/kernel/qapplication_p.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/widgets/kernel/qapplication.cpp b/src/widgets/kernel/qapplication.cpp
index f5e7a3a412e..f4eba61a80e 100644
--- a/src/widgets/kernel/qapplication.cpp
+++ b/src/widgets/kernel/qapplication.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #include "qplatformdefs.h"
 #include "qabstracteventdispatcher.h"
diff --git a/src/widgets/kernel/qapplication.h b/src/widgets/kernel/qapplication.h
index c2c5f7264d4..b56bcec294a 100644
--- a/src/widgets/kernel/qapplication.h
+++ b/src/widgets/kernel/qapplication.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QAPPLICATION_H
 #define QAPPLICATION_H
diff --git a/src/widgets/kernel/qapplication_p.h b/src/widgets/kernel/qapplication_p.h
index 3e7c5edab45..2b33e33a2a4 100644
--- a/src/widgets/kernel/qapplication_p.h
+++ b/src/widgets/kernel/qapplication_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2016 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant reason:default
 
 #ifndef QAPPLICATION_P_H
 #define QAPPLICATION_P_H
-- 
2.51.2

